src/odb/src/lefin/lefin.cpp

```diff
@@ -388,15 +394,17 @@ bool lefin::addGeoms(dbObject* object, bool is_pin, lefiGeometries* geometry)
         p.x = pItr->x;
         p.y = pItr->y;
 
-        for (y = 0; y < pItr->yStart; y++)
-          for (x = 0; x < pItr->xStart; x++)
+        for (y = 0; y < pItr->yStart; y++) {
```

Comment:
This does look like an actual problem.

Reply:
The LEF syntax is
```
DO numX BY numY STEP spaceX spaceY
```
but that seems to turn into 
```
  double xStart;
  double yStart;
  double xStep;
  double yStep;
```
which is quite odd.   I think xStart==numX (likewise Y) so it should be safe to just cast it to an int.  I've never seen this syntax actually used.

Reply:
Theoretically it's supposed to be used for PDN as a compact representation of repeated structures. It would be good to take advantage of it.

It makes it hard to do any kind of geometric queries on the PDN though.